src/pages/resultsView/ResultsViewPage.tsx

```diff
@@ -433,13 +433,17 @@ export default class ResultsViewPage extends React.Component<IResultsViewPagePro
         } else {
             return (<>
                 {
-                    // if query invalid(we only check gene length for now), return error page
+                    // if query invalid(we only check gene count * sample count < 1,000,000 for now), return error page
                     (this.resultsViewPageStore.isQueryInvalid) && (
-                        <div className="alert alert-danger queryInvalid" style={{marginBottom: "15px"}} role="alert">
+                        <div className="alert alert-danger queryInvalid" style={{marginBottom: "40px"}} role="alert">
                             <p>
-                                Queries are limited to 100 genes. Please <a
-                                href={`mailto:${AppConfig.serverConfig.skin_email_contact}`}>let us know</a> your use
-                                case(s) if you need to query more than 100 genes.
+                                Queries are limited by gene and sample count. A query must either have up to {AppConfig.serverConfig.query_gene_limit.toLocaleString()} genes or
```

Comment:
Lets reverse the language here:   "A query is INVALID if gene count x sample count exceeds ${limit}"